cli/flox/src/commands/auth.rs

```diff
@@ -59,6 +60,10 @@ fn create_oauth_client() -> Result<BasicClient> {
 }
 
 pub async fn authorize(client: BasicClient) -> Result<Credential> {
+    if !Dialog::can_prompt() {
+        bail!("Cannot prompt for user input")
```

Comment:
nonblocking question: do we support non-interactive authorization?

Reply:
yes, by setting a token via the config

`FLOX_FLOXHUB_TOKEN=<token>` or `flox config --set floxhub_token <token>`

I don't know whether we have a way to _get_ that token from within the SaaS that's a question for @nicklewis I guess, but not related to this PR/issue.

Reply:
No need to add here, but at some point we should tell users what they can do to login non-interactively instead of just bailing